phoenix-core/src/main/java/org/apache/phoenix/coprocessor/UngroupedAggregateRegionScanner.java

```diff
@@ -609,6 +613,9 @@ public boolean next(List<Cell> resultsToReturn) throws IOException {
                     } while (hasMore && (EnvironmentEdgeManager.currentTimeMillis() - startTime) < pageSizeInMs);
 
                     if (!mutations.isEmpty()) {
+                        if (isDelete || isUpsert) {
+                            annotateDataMutations(mutations, scan);
+                        }
```

Comment:
It would be nice to refactor this perhaps to a new method? Lines 596 to 602 is same as 616 to 622

Reply:
Done